### ISSUE TITLE
Other micro tasks can be executed while initPolyfill is running

### ISSUE THREAD

**Issue by Artur-** [calebdwilliams/construct-style-sheets#76]
**Describe the bug**
This code in `initPolyfill` can be interrupted in the middle: https://github.com/calebdwilliams/construct-style-sheets/blob/b793dd74c7b1b4c03617471c041712194a55a467/src/init.js#L20-L25

It can be seen by running e.g. the following in a browser console
```
queueMicrotask(() => {
  console.log('micro task');
});
console.log('creating iframe');
var iframe = document.createElement('iframe');
document.body.appendChild(iframe);
console.log('iframe appended');
```

In Firefox you get
```
creating iframe
iframe appended
micro task
```

But in Safari you get
```
creating iframe
micro task
iframe appended
```

Is this an academic problem? Unfortunately not. 

I see the problem in my application using Lit 2 + this polyfill. It so happens that rendering a Lit element triggers loading of the polyfill and scheduling rendering of that element. As a result of rendering the element, `document.adoptedStyleSheets` is used but fails when `instanceOfStyleSheet` in the polyfill is called as the polyfill is in use but only half-initialized
```
TypeError: Right hand side of instanceof is not an object
```
because `frame.CSSStyleSheet` is not defined yet. It is set on the row following `document.body.appendChild(iframe);` in `initPolyfill`

Interestingly enough, I see the issue also in Firefox even though the simple micro task test case indicates I should not.

**To Reproduce**
Wasn't able to extract a test case.

**Expected behavior**
Initialization of the polyfill would be non-interruptible so this would be a non issue.

**Comment by calebdwilliams** [comment]
Interesting. Could you push out a reproduction repo, CodePen or StackBlitz? I’d love to poke around at this and also see how #72 would affect this. I’ve had spotty availability this week, but am hoping to review/merge that next week. If this bug is fixed in that version, I will prioritize getting that merged/released.

**Comment by Artur-** [comment]
I wasn't able to produce a reliable test case but I can give it another shot

**Comment by Artur-** [comment]
Created https://stackblitz.com/edit/js-3fovpz?file=index.js 
At least in this case it requires an invalid value to be assigned to `document.adoptedStyleSheets` to the problematic instanceof check is triggered

**Comment by Lodin** [comment]
@Artur-, could you please check if the issue is resolved with the `3.0.0-0` version?

**Comment by Artur-** [comment]
I no longer see `TypeError: Right hand side of instanceof is not an object` with 3.0.3 🎉
